canon-docs/src/snippets/button.tsx

```diff
@@ -1,102 +1,18 @@
 'use client';
 
-import { Button, Flex, ButtonProps, Text } from '../../../packages/canon';
+import { composeStories } from '@storybook/react';
+import * as ButtonStories from '../../../packages/canon/src/components/Button/Button.stories';
 
-export const ButtonPreview = () => {
-  return (
-    <Flex align="center">
-      <Button iconStart="cloud" variant="primary">
-        Button
-      </Button>
-      <Button iconStart="cloud" variant="secondary">
-        Button
-      </Button>
-    </Flex>
-  );
-};
-
-export const ButtonSizes = () => {
-  return (
-    <Flex align="center">
-      <Button size="medium">Medium</Button>
-      <Button size="small">Small</Button>
-    </Flex>
-  );
-};
-
-export const ButtonWithIcons = () => {
-  return (
-    <Flex align="center">
-      <Button iconStart="cloud">Button</Button>
-      <Button iconEnd="chevron-right">Button</Button>
-      <Button iconStart="cloud" iconEnd="chevron-right">
-        Button
-      </Button>
-    </Flex>
-  );
-};
-
-export const ButtonFullWidth = () => {
-  return (
-    <Flex direction="column" style={{ width: '300px' }}>
-      <Button iconStart="cloud">Button</Button>
-      <Button iconEnd="chevron-right">Button</Button>
-      <Button iconStart="cloud" iconEnd="chevron-right">
-        Button
-      </Button>
-    </Flex>
-  );
-};
-
-export const ButtonDisabled = () => {
-  return <Button disabled>Button</Button>;
-};
-
-export const ButtonResponsive = () => {
-  return (
-    <Button variant={{ initial: 'primary', lg: 'secondary' }}>
-      Responsive Button
-    </Button>
-  );
-};
+export const ButtonSnippet = ({ story }: { story: string }) => {
+  const stories = composeStories(ButtonStories);
 
-export const ButtonPlayground = () => {
-  const variants: string[] = ['primary', 'secondary', 'tertiary'];
+  if (story === 'Variants') return <stories.Variants />;
+  if (story === 'Sizes') return <stories.Sizes />;
+  if (story === 'WithIcons') return <stories.WithIcons />;
+  if (story === 'FullWidth') return <stories.FullWidth />;
```

Comment:
How complex is the `stories` object? Could you `const StoryComponent = stories[story]; return StoryComponent ? <StoryComponent/> : null` instead of the `if`s?

Reply:
Done